include/s3select_oper.h

```diff
@@ -1535,43 +1547,70 @@ class logical_operand : public base_statement
       throw base_s3select_exception("missing operand for logical ", base_s3select_exception::s3select_exp_en_t::FATAL);
     }
     value a = l->eval();
+    auto abool = a.is_bool() ? 0 : 1;
```

Comment:
once the operator=(bool) is changed
maybe there is no need for all these changes